### PR TITLE
[Intel GPU] Dispatch Stub support

### PR DESCRIPTION
**Pull request by ZhiweiYan-96**
# Motivation
Structured codegen is beneficial for easier decoupling tensor meta setting and kernel implementation. At present, XPU operators need to handle tensor metas in hand-written way. 

We plan to leverage the codegen system for auto generate structured operators. This PR facilitate the `DispatchStub` support for  Intel GPUs. Based on that, XPU operators would have possibility to register kernel functor to operator stubs.

This is a prerequisite of PR #127865, where we will modify the codegen system to generate XPU needed source files and headers.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #129452
* #127865
* __->__ #127860

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @chauhang @d4l3k @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @mcarilli @ptrblck @leslie-fang-intel @EikanWang @voznesenskym @penguinwu @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @LucasLLC @MeetVadakkanchery @mhorowitz @gujinghui @fengyuan14 @guangyey

### CODE REVIEW

File: aten/src/ATen/native/DispatchStub.cpp [hunk 1]
```diff
@@ -112,7 +112,8 @@ DispatchResult DispatchStubImpl::try_get_call_ptr(
         c10::DeviceType::CUDA,
         c10::DeviceType::HIP,
         c10::DeviceType::MPS,
-        c10::DeviceType::PrivateUse1
+        c10::DeviceType::PrivateUse1,
+        c10::DeviceType::XPU
```
Comment (gujinghui) [inline aten/src/ATen/native/DispatchStub.cpp:116]: The PrivateUse1 should be always at the bottom?

Comment (ZhiweiYan-96) [reply aten/src/ATen/native/DispatchStub.cpp]: sure, has modified.

**Review by gujinghui — APPROVED** [review]

**Review by EikanWang — APPROVED** [review]

**Comment by EikanWang** [comment]
Please add test cases.

**Comment by EikanWang** [comment]
@albanD , @atalman , @malfet , may I know if you have any comments on this PR?

**Pull request closed without merging**